### PR TITLE
Vulnerability notification example

### PR DESCRIPTION
**Pull request by saturnism**
Fixes #issue

> It's a good idea to open an issue first for discussion.

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [x] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: container-registry/vulnerability-notification-function/.mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/*
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/.mvn/wrapper/MavenWrapperDownloader.java:1]: We try to stay away from this stuff.

File: container-registry/vulnerability-notification-function/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/LICENSE:1]: Is there a reason you are putting a seperate license here?  It isn't nescessary or appropiate.

File: container-registry/vulnerability-notification-function/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## Detailed Steps
+Enable Container Analysis:
+```
+gcloud services enable containeranalysis.googleapis.com
+```
+
+Build and Deploy this function:
+```
+./mvnw package
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/README.md:25]: just use `mvn`, assume they have the right version.  It's how our things work.

File: container-registry/vulnerability-notification-function/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  --policy-from-file=cloudops/vulnerability-policy.yaml
+```
+
+This configuration doesn't configure any notification channels. To get notified via e-mail, or Slack
+see [Manging notification channels documentation](https://cloud.google.com/monitoring/support/notification-options).
+
+## Development
+Run Locally:
+```
+./mvnw function:run
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/README.md:52]: `mvn`

File: container-registry/vulnerability-notification-function/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+Build:
+```
+./mvnw clean package
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/README.md:57]: `mvn`

File: container-registry/vulnerability-notification-function/cloudops/container-analysis-metrics.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  labels:
+    - description: CVE
+      key: cve
+    - description: Severity
+      key: severity
+    - description: Image Name
+      key: image
+  metricKind: DELTA
+  unit: '1'
+  valueType: INT64
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/cloudops/container-analysis-metrics.yaml:19]: does this need a newline?

File: container-registry/vulnerability-notification-function/cloudops/vulnerability-policy.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      comparison: COMPARISON_GT
+      duration: 60s
+      filter: metric.type="logging.googleapis.com/user/container-vulnerability" resource.type="cloud_function"
+      trigger:
+        count: 1
+    displayName: Container Vulnerability by Image [COUNT]
+documentation:
+  content: Container Image ${metric.label.image} has N vulnerabilities!
+  mimeType: text/markdown
+enabled: true
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/cloudops/vulnerability-policy.yaml:20]: newline?

File: container-registry/vulnerability-notification-function/mvnw [hunk 8]
```diff
@@ -0,0 +1,310 @@
+#!/bin/sh
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/mvnw:1]: skip this.

File: container-registry/vulnerability-notification-function/mvnw.cmd [hunk 9]
```diff
@@ -0,0 +1,182 @@
+@REM ----------------------------------------------------------------------------
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/mvnw.cmd:1]: skip this

File: container-registry/vulnerability-notification-function/pom.xml [hunk 10]
```diff
@@ -0,0 +1,90 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/pom.xml:1]: license?

File: container-registry/vulnerability-notification-function/pom.xml [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>libraries-bom</artifactId>
+        <version>8.1.0</version>
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/pom.xml:69]: a bit old

File: container-registry/vulnerability-notification-function/pom.xml [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>com.google.cloud.functions</groupId>
+        <artifactId>function-maven-plugin</artifactId>
+        <version>0.9.5</version>
```
Comment (lesv) [inline container-registry/vulnerability-notification-function/pom.xml:81]: this may be old as well.

**Review by lesv — CHANGES_REQUESTED** [review]
PTAL

File: container-registry/vulnerability-notification-function/pom.xml (hunk 12 above)
Comment (lesv) [reply container-registry/vulnerability-notification-function/pom.xml:81]: actually it's ok

File: container-registry/vulnerability-notification-function/mvnw.cmd (hunk 9 above)
Comment (saturnism) [reply container-registry/vulnerability-notification-function/mvnw.cmd:1]: this is standard maven wrapper.

File: container-registry/vulnerability-notification-function/cloudops/container-analysis-metrics.yaml (hunk 6 above)
Comment (saturnism) [reply container-registry/vulnerability-notification-function/cloudops/container-analysis-metrics.yaml:19]: not needed.

File: container-registry/vulnerability-notification-function/.mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (lesv) [reply container-registry/vulnerability-notification-function/.mvn/wrapper/MavenWrapperDownloader.java:1]: The maven wrapper shouldn't be in our samples.

File: container-registry/vulnerability-notification-function/mvnw.cmd (hunk 9 above)
Comment (lesv) [reply container-registry/vulnerability-notification-function/mvnw.cmd:1]: Please delete the maven wrapper - we don't use them in our samples

**Review by lesv — COMMENTED** [review]
Please remove the maven wrapper and the license

File: container-registry/vulnerability-notification-function/.mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (saturnism) [reply container-registry/vulnerability-notification-function/.mvn/wrapper/MavenWrapperDownloader.java]: removed

File: container-registry/vulnerability-notification-function/LICENSE (hunk 2 above)
Comment (saturnism) [reply container-registry/vulnerability-notification-function/LICENSE]: copied over from the original repo. removed.

File: container-registry/vulnerability-notification-function/cloudops/vulnerability-policy.yaml (hunk 7 above)
Comment (saturnism) [reply container-registry/vulnerability-notification-function/cloudops/vulnerability-policy.yaml:20]: not needed here.

**Review by lesv — APPROVED** [review]

**Pull request merged**